ludwig/utils/torch_utils.py

```diff
@@ -106,13 +106,14 @@ def input_dtype(self):
     @property
     @abstractmethod
     def input_shape(self) -> torch.Size:
-        """ Returns the size of the input tensor without the batch dimension. """
+        """ Returns size of the input tensor without the batch dimension."""
         raise NotImplementedError('Abstract class.')
 
     @property
+    @abstractmethod
     def output_shape(self) -> torch.Size:
-        """ Returns the size of the output tensor without the batch dimension."""
-        return self._compute_output_shape()
+        """ Returns size of the output tensor without the batch dimension."""
+        raise NotImplementedError('Abstract class.')
```

Comment:
here we want to keep that _compute_output_shape as default behavior and submodules can override it if they want

Reply:
Done.